lib/system/io.nim

```diff
@@ -186,6 +186,21 @@ proc readChars*(f: File, a: var openArray[char], start, len: Natural): int {.
   ## reads `len` bytes into the buffer `a` starting at `a[start]`. Returns
   ## the actual number of bytes that have been read which may be less than
   ## `len` (if not as many bytes are remaining), but not greater.
+  ##
+  ## .. code-block:: nim
```

Comment:
Please use `runnableExamples` instead of `.. code-block:: nim`. It has the advantage, that the code actually compiled and run (or you do `runnableExamples("-r:off")` to only compile it).

Reply:
I was going to but this example isn't runnable as it requires a path to a real file; unless I misunderstand how runnableExamples work?

Reply:
As I said, you can use `runnableExamples("-r:off")` if the example should only be compiled and not run.

Reply:
Ah I see.